src/Microsoft.AspNet.Mvc.ModelBinding/Internal/ModelBindingHelper.cs

```diff
@@ -100,7 +100,20 @@ internal static void AddModelErrorBasedOnExceptionType(ModelBindingContext bindi
         {
             if (IsFormatException(ex))
             {
-                bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
+                var modelState = bindingContext.ModelState[bindingContext.ModelName];
+                if (modelState == null)
+                {
+                    bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
```

Comment:
current invariant is that `Message==null` when `Exception!=null`.  means we never attempt to display an error message for exceptions caught during model binding / validation.